socketioxide/src/adapter.rs

```diff
@@ -274,6 +280,15 @@ impl Adapter for LocalAdapter {
         Ok(())
     }
 
+    fn get_socket(&self, sid: Sid) -> Option<SocketRef<Self>>
+    where
+        Self: Sized,
+    {
+        self.apply_opts(BroadcastOptions::new(Some(sid)))
+            .into_iter()
+            .find(|socket| socket.id == sid)
+    }
+
```

Comment:
The get_socket fn should not rely on the adapter because sockets are local (therefore your implementation is only looping back to the ns.get_socket fn).

Therefore the fn should directly call get_socket() on the ns.